python-checks/src/main/java/org/sonar/python/checks/UnusedLocalVariableCheck.java

```diff
@@ -64,4 +71,23 @@ private static boolean isCallingLocalsFunction(PyFunctionDefTree functionTree) {
       .map(PyCallExpressionTree::callee)
       .anyMatch(callee -> callee.is(Kind.NAME) && "locals".equals(((PyNameTree) callee).name()));
   }
+
+  private static Set<String> extractStringInterpolationIdentifiers(PyFunctionDefTree functionTree) {
+    return functionTree.descendants(Kind.STRING_LITERAL)
+      .map(PyStringLiteralTree.class::cast)
+      .flatMap(str -> str.stringElements().stream())
+      .filter(str -> str.prefix().equals("f"))
+      .map(PyStringElementTree::trimmedQuotesValue)
+      .flatMap(UnusedLocalVariableCheck::extractInterpolations)
+      .collect(Collectors.toSet());
+  }
+
+  private static Stream<String> extractInterpolations(String str) {
+    Matcher matcher = INTERPOLATION_PATTERN.matcher(str);
+    List<String> identifiers = new ArrayList<>();
+    while (matcher.find()) {
+      identifiers.add(matcher.group(1));
```

Comment:
What about the following code?
```
x = ''
print(f"{'}' + x}")
```

Reply:
I would expect this to happen not so often. I'd suggest to have a ticket to create `F-string` in the AST and handle this case properly, wdyt?
I'm adding a test for this FP anyway.

Reply:
[Ticket](https://jira.sonarsource.com/browse/SONARPY-425) created